packages/signers/pjs-signer/src/from-pjs-account.ts

```diff
@@ -24,13 +22,8 @@ export const getAddressFormat = (metadata: V14 | V15): number => {
   return dynamicBuilder.buildDefinition(constant.type).dec(constant.value)
 }
 
-const versionCodec = enhanceEncoder(
-  u8.enc,
-  (value: { signed: boolean; version: number }) =>
-    (+!!value.signed << 7) | value.version,
-)
-
-const getPublicKey = AccountId().enc
+const getPublicKey = (address: string) =>
+  address.startsWith("0x") ? fromHex(address) : AccountId().enc(address)
```

Comment:
We could maybe use the hex regex instead to ensure that addresses without `0x` pass as well. There is a very little chance that a polkadot address passes the test though.